worlds/tloz_oos/__init__.py

```diff
@@ -634,6 +645,23 @@ def build_item_pool_dict(self):
         item_pool_dict["Gasha Seed"] = required_gasha_seeds
         filler_item_count -= required_gasha_seeds
 
+        # Add the required rings
+        ring_copy = sorted(self.options.required_rings.value.copy())
+        for _ in range(len(ring_copy)):
+            ring_name = ring_copy.pop()
+            if ITEMS_DATA[ring_name]["classification"] == ItemClassification.filler:
```

Comment:
This feels weird, I feel either that check should be done on the !USEFUL handling or !USEFUL should add the useful flag instead of replacing it